### PR TITLE
Add organization profile editing functionality

### PR DESCRIPTION
**Pull request by elazdi-al**
> [!IMPORTANT]
> 2852 insertions(+), 12 deletions(-) are due to testing and adapting tests
> PR documentation, description and tests were done by an AI for better readability and maintainability 

## What?

This PR adds the ability for organization owners to edit their organization profiles directly within the app. It introduces a new edit screen that allows owners to modify organization details including name, type, description, location, social links, logo, and manage members.

## Why?

Previously, organization owners had no way to update their organization information after creation. This made it impossible to:
- Correct mistakes in organization details
- Update social media links as they change
- Modify the organization logo or description
- Manage member roles or remove members

This feature is essential for maintaining accurate and up-to-date organization information in the app.

## How?

**Implementation:**
- Created `OrganizationProfileEditScreen` with a comprehensive UI for editing all organization fields
- Implemented `OrganizationProfileEditViewModel` to handle state management and business logic
- Added validation for required fields (name, description length limits)
- Integrated logo upload functionality with Firebase Storage
- Added member management UI with role modification and member removal capabilities
- Added navigation from OrganizationManagementScreen to the edit screen for owners

**Key Features:**
- Real-time validation with error messages
- Logo upload with preview and remove functionality
- Organization type selector with visual feedback
- Member list with role badges and action buttons
- Owner protection (cannot be removed or have role changed)
- Success/error snackbar notifications

**Testing:**
- Added 37 comprehensive unit tests for the ViewModel (100% coverage)
- Added 16 UI tests for the edit screen
- Added 3 navigation route tests
- All 480 organization-related tests pass
- Tested error scenarios, edge cases, and validation logic
image

### CODE REVIEW

File: app/src/main/java/com/github/se/studentconnect/ui/screen/profile/OrganizationProfileEditScreen.kt [hunk 1]
```diff
@@ -0,0 +1,909 @@
...
+                                fontWeight = FontWeight.Bold,
+                                color = MaterialTheme.colorScheme.onPrimaryContainer)
+                          }
+                    }
+              }
+            }
+      }
+}
+
+/** Enhanced remove member confirmation dialog. */
```
Comment (haaissa) [inline app/src/main/java/com/github/se/studentconnect/ui/screen/profile/OrganizationProfileEditScreen.kt]: poor doc

**Review by haaissa — CHANGES_REQUESTED** [review]
Nice job! Sonar line coverage passes, 0 issues and 0% duplications. I requested just quick changes.

> [!IMPORTANT]
> Add these information
image

> [!WARNING]
> Add image to your PR description of the new screen

Good work overall as usual
